### PR TITLE
Only unique certs

### PR DESCRIPTION
**Pull request by DieMyst**

### CODE REVIEW

File: src/NetworkMap/Certificates/Update.elm [hunk 1]
```diff
@@ -16,20 +16,101 @@ limitations under the License.
...
+        chain =
+            cert.chain
+
+        ( _, pairs ) =
+            chain |> Array.foldl gatherPairs ( Nothing, [] )
+    in
+    pairs
+
+
+gatherPairs : Trust -> ( Maybe String, List ( String, String, Trust ) ) -> ( Maybe String, List ( String, String, Trust ) )
```
Comment (alari) [inline src/NetworkMap/Certificates/Update.elm:42]: Maybe define a type for `(String, String, Trust)`?

`gather` doesn't give any insight into what happens there.

File: src/NetworkMap/Certificates/View.elm [hunk 2]
```diff
@@ -30,12 +31,36 @@ import NetworkMap.Certificates.Msg exposing (Msg(..))
...
+                            a
+            )
+            A.empty
 
 
 view : Screen.Model -> Model -> Element Msg
 view screen networkModel =
+    let
+        maybes =
+            networkModel.certificates |> A.map (\{ trustIds } -> trustIds |> A.map (\p -> Dict.get p networkModel.trusts))
```
Comment (alari) [inline src/NetworkMap/Certificates/View.elm:57]: Have you applied the formatter? Lines are pretty long in this PR.

File: src/NetworkMap/Certificates/View.elm [hunk 3]
```diff
@@ -30,12 +31,36 @@ import NetworkMap.Certificates.Msg exposing (Msg(..))
 import Palette exposing (fillWidth, limitLayoutWidth, shortHashRaw)
 import Screen.Model as Screen
 import Time
+import Utils.ArrayExtras as ArrayExtras
+
+
+flatMaybes : Array (Maybe a) -> Array a
```
Comment (alari) [inline src/NetworkMap/Certificates/View.elm:37]: Maybe move it to Array's Extras?

File: src/NetworkMap/Certificates/Update.elm [hunk 4]
```diff
@@ -16,20 +16,101 @@ limitations under the License.
...
+import List
+import List.FlatMap
+import List.Unique
+import NetworkMap.Certificates.Model exposing (Certificate, CertificateIds, Model, Trust)
 import NetworkMap.Certificates.Msg exposing (Msg(..))
 import NetworkMap.Certificates.Port as Port
+import Utils.ArrayExtras as ArrayExtras
+
+
+convertCert : Certificate -> List ( String, String, Trust )
```
Comment (alari) [inline src/NetworkMap/Certificates/Update.elm:30]: Convert into _what_?

File: src/NetworkMap/Certificates/View.elm (hunk 2 above)
Comment (DieMyst) [reply src/NetworkMap/Certificates/View.elm:57]: Yes, I applied. I'll recheck it.

**Pull request merged**